content/Watsonx/WatsonxAI/102.md

```diff
@@ -0,0 +1,129 @@
+---
+title: '102: Programmatic Access'
+timeToComplete: 30
+updated: 2023-09-19
+---
+
+# 102: Programmatic Access to watsonx.ai
+
+In this lab we'll explore two ways that we can programmatically access watsonx.ai. We will do this through the use of some open source tools like Python and Jupyter Notebooks.
+
+First though, we'll need to acquire some prerequisites from your IBM Cloud account in order to access the APIs
+
+## Getting prerequisites
+
+### Creating an API Key
+
+In order to authenticate with watsonx externally via an API you will first need to create an API key that's tied to your IBM Cloud Account.
+
+To do this:
+
+1. Start by heading to your [cloud account's dashboard](https://cloud.ibm.com/)
+2. Click **Manage** at the top of the page, then **Access (IAM)**
+
+![to_iam](./images/102/1.png)
+
+3. In the left-hand navigation menu click **API Keys**
+
+![iam_page](./images/102/2.png)
+
+4. A table should appear with a list of your current API keys. Click the **Create +** button on the top-right of the table
+
+![api_keys](./images/102/3.png)
+
+5. In the menu that appears given your API key a name and an optional description and click the **Create** button
+
+![create_key](./images/102/4.png)
+
+6. Lastly, A new menu will appear titled **API key successfully created**. Click either the **Copy** or the **Download** button. If clicking copy be sure to immediately store this value in a text file for later use.
+
+![copy_key](./images/102/5.png)
+
+With that you have a valid API key.
+
+### Getting your project ID
+
+We now need to find the project ID of your watsonx instance. Start by navigating to the overview page for watsonx [here](https://cloud.ibm.com/watsonx/overview). Once there:
+
+1. Click the **Launch** button inside the **watsonx.ai** card on the page. This will open a new tab.
+
+![launch_ai](./images/102/6.png)
+
+2. Once in the new tab you should see your project under the **Projects** list in the **Recent work** section. Go ahead and click the relevant project.
+
+![recent_project](./images/102/7.png)
+
+3. In the project overview section click the **Manage** tab at the top of the page
```

Comment:
It must cache which LHN item you previously had open.  When I clicked on Manage I saw the "Access control" screen rather than "General".  Can we add on the end of this step `and choose "General" from the left hand navigation menu.`